File: RecipeChangeTracker.Models/Recipe.cs
using System;
using System.Collections.Generic;

namespace RecipeChangeTracker.Models
{
    public class Recipe
    {
        public string Name { get; }

        public IEnumerable<Ingredient> Ingredients { get; }

        public IList<string> Steps { get; }

        public TimeSpan CookTime { get; }

        public Recipe(string name, TimeSpan cookTime, IEnumerable<Ingredient> ingredients, IList<string> steps)
        {
            Name = name;
            Ingredients = ingredients;
            Steps = steps;
            CookTime = cookTime;
        }
    }
}

File: RecipeChangeTracker/Functions/Recipe.fs
namespace RecipeChangeTracker.Functions

open RecipeChangeTracker.Types
open System

module Recipe =

    let create name ingredients steps cooktime =
        { Name = name
          Ingredients = ingredients
          Steps = steps
          CookTime = cooktime }

    let createEmpty =
        create ""  [] [] TimeSpan.Zero
